scripts/generate_dump

```diff
@@ -144,21 +155,24 @@ save_bcmcmd_all_ns() {
 #  None
 ###############################################################################
 save_cmd() {
+    trap 'handle_error $? $LINENO' ERR
     local start_t=$(date +%s%3N)
     local end_t=0
     local cmd="$1"
     local filename=$2
     local filepath="${LOGDIR}/$filename"
     local do_gzip=${3:-false}
-    local save_stderr=${4:-true}
+    local save_stderr=${4:-$SAVE_STDERR}
```

Comment:
>SAVE_STDERR [](http://example.com/codeflow?start=28&length=11)

You should add `$SAVE_STDERR` to caller place as an argument, and keep `true` as default parameter value.

Reply:
From what i understand, wouldn't that require adding an extra argument to all the save_cmd invocations across the script? That would be fine if this argument is something like `filename` which might change for different invocations.

But since the scope of this flag is global, i think it makes sense to have it inside the parameter value. Let me know if you think otherwise

Reply:
I could understand your point. Does it make more sense to only use global variable and remove `$4` ?

Reply:
The stderr arg was first introduced here https://github.com/Azure/sonic-utilities/pull/1335. 

And for some reason, this was set to false in the normal execution itself (probably because he only wanted to collect stdout). I did not want to disturb it and thus retained the argument, but if not provided read from the global variable.

Reply:
Could you discuss with @stepanblyschak in the same company to understand the original intention? To me, use a global variable as default value is weird.

Reply:
@stepanblyschak, confirmed that this local variable $4 can be removed. I'll remove it and will only have the global variable.

Reply:
DONE